shared/components/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,116 @@
+import { useForm, SubmitHandler, Controller } from "react-hook-form";
+import { Button, Modal, Select, TextInput, Textarea } from "@mantine/core";
+import { DateInput } from "@mantine/dates";
+import "@mantine/dates/styles.css";
+
+import { TTask } from "@/shared/typedefs/types";
+import { EPriority } from "@/shared/typedefs/enums";
+
+import { ITaskModalProps } from "./TaskModal.type";
+
+const TaskModal = ({
+  opened,
+  close,
+  title,
+  buttonTitle,
+  task,
+  onSubmitAction,
+  resetAfterSubmit,
+}: ITaskModalProps) => {
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, isValid },
+    reset,
+  } = useForm<TTask>({ mode: "onBlur" });
+
+  const validateDueDate = (value: Date | null) => {
+    const currentDate = new Date();
+
+    if (value && value < currentDate) {
+      return "Due date cannot be in the past";
+    }
+
+    return true;
+  };
+
+  const onSubmit: SubmitHandler<TTask> = (data: TTask) => {
+    onSubmitAction && onSubmitAction(data);
+    resetAfterSubmit && reset();
+  };
+
+  return (
+    <Modal opened={opened} onClose={close} title={title} centered>
+      <form
+        onSubmit={handleSubmit(onSubmit)}
+        style={{ display: "flex", flexDirection: "column", gap: 12 }}
+      >
```

Comment:
Why are we only validating due date? Please add a ticket to address form validation.